File: resources/DataTableModel.java
import com.google.gson.JsonArray;
import com.google.gson.JsonObject;

import java.util.ArrayList;
import java.util.List;

import javax.swing.table.AbstractTableModel;

public class DataTableModel extends AbstractTableModel {

    private JsonArray data = new JsonArray();
    private String[] columnNames = null;

    public DataTableModel(JsonArray data) {
        this.data = data;
        JsonObject obj = data.get(0).getAsJsonObject();
        columnNames = obj.keySet().toArray(new String[0]);
    }

    public DataTableModel(JsonArray data, String[] columnNames) {
        this.data = data;
        this.columnNames = columnNames;
    }

    public String[] getColumnNames() {
        return columnNames;
    }

    public void setColumnNames(String[] columnNames) {
        this.columnNames = columnNames;
    }

    @Override
    public String getColumnName(int column) {
        return columnNames[column];
    }

    @Override
    public int getColumnCount() {
        return columnNames.length;
    }

    @Override
    public int getRowCount() {
        return data.size();
    }

    @Override
    public Object getValueAt(int row, int column) {
        Object userAttribute = null;
        JsonObject object = data.get(row).getAsJsonObject();
        return object.get(columnNames[column]);
    }
}

File: src/model/DataTableModel.kt
package model

import com.google.gson.JsonArray
import com.google.gson.JsonElement
import com.google.gson.JsonObject
import javax.swing.table.AbstractTableModel

class DataTableModel : AbstractTableModel {
    private var data = JsonArray()
    var columnNames: Array<String>? = null

    constructor(data: JsonArray) {
        this.data = data

        val obj = data[0].asJsonObject
        columnNames = obj.keySet().toTypedArray()
    }

    constructor(data: JsonArray, columnNames: Array<String>?) {
        this.data = data
        if (columnNames.isNullOrEmpty()){
            val obj = data[0].asJsonObject
            this.columnNames = obj.keySet().toTypedArray()
        }
        this.columnNames = columnNames
    }

    override fun getColumnName(column: Int): String {
        return columnNames!![column]
    }

    override fun getColumnCount(): Int {
        return columnNames!!.size
    }

    override fun getRowCount(): Int {
        return data.size()
    }

    override fun getValueAt(row: Int, column: Int): Any {
        val obj = data[row].asJsonObject
        try {
            return obj.get(columnNames!![column]).asString
        }
        catch (_ : Exception){
            return obj.get(columnNames!![column])
        }
    }

    fun getValue(row: Int, column: String) : Any{
        val obj = data[row].asJsonObject
        try {
            return obj.get(column).asInt
        }
        catch (_: Exception){}
        try {
            return obj.get(column).asString
        }
        catch (e : Exception){
            return obj.get(column)
        }
    }

    fun getObject(row: Int): Any {
        return data[row].asJsonObject
    }
}